### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by nathanshelly**
Adds unit tests for Board, GameUtility and RuleEnforcer (more should be added in future)

### CODE REVIEW

**Review by vkoves — COMMENTED** [review]
@nathanshelly - can you update the README to indicate how you can run these tests?

I would also suggest integrating a tool like [Travis](https://travis-ci.org/) to automatically run tests for us, but that seems outside the scope of this PR and can be made into an issue.

**Comment by vkoves** [comment]
For others, `make test` is how you run tests. We should add that to the README and make sure we mention that folk should run `npm install`, as that's definitely needed.

File: test/board.test.ts [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
+  it("should have expected number of colors with expected distribution", () => {
+    let board = new Board(Team.blue);
+    
+    let num_red = board.colors.filter(n => n === Color.red).length;
+    let num_blue = board.colors.filter(n => n === Color.blue).length;
+    let num_neutral = board.colors.filter(n => n === Color.neutral).length;
+    let num_assassins = board.colors.filter(n => n === Color.assassin).length;
+    
+    expect(board.colors.length).to.equal(25);
+    expect(num_red).to.be.within(8, 9);
```
Comment (vkoves) [inline test/board.test.ts]: Since the starting team is blue, wouldn't blue always have 9 cards, and red always 9? Testing this myself showed this to be a correct assumption, so I'd suggest changing this test to specify those and documenting why that is. Alternatively, you could make tests that verify both blue and red having 9 cards if they are the current team.

File: test/game_utility.test.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+		let red_p_one = new Spymaster("red", "1", Team.red, mock_ws_instance);
+		let red_p_two = new Operative("red", "2", Team.red, mock_ws_instance);
+		let red_p_three = new Operative("red", "3", Team.red, mock_ws_instance);
+		let blue_p_one = new Spymaster("blue", "1", Team.blue, mock_ws_instance);
+		let blue_p_two = new Operative("blue", "2", Team.blue, mock_ws_instance);
+		players = [red_p_one, red_p_two, red_p_three, blue_p_one, blue_p_two];
+	});
+
+  it("should correctly split loiterers on team", () => {
```
Comment (vkoves) [inline test/game_utility.test.ts:39]: Make sure to consistently use one type of indenting. Tab based indentation is used everywhere else in this file, but this line (and a few others) uses two spaces.

File: test/rule_enforcer.test.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	});
+
+	it("should not allow words not in dictionary", () => {
+		expect(re.isValidWord("bae")).to.be.false
+		expect(re.isValidWord("porg")).to.be.false;
+		expect(re.isValidWord("asdfasdjflasdjflkasdjflkasdj")).to.be.false;
+		expect(re.isValidWord("supercalifragilistcexpialidocious")).to.be.false;
+	});
+
+	it("should return false for words that are not on the board (regardless of dictionary)", () => {
```
Comment (vkoves) [inline test/rule_enforcer.test.ts]: The rest of these tests describe the enforcer allowing or not allowing words. This makes more sense than returning true or false, which is specific to `wordOnBoard`. In this case, I'd probably rephrase this to say `wordOnBoard should return false...` so that it is clear the general enforcer should not be disallowing words because they aren't on the board.

**Review by vkoves — COMMENTED** [review]
I have a few specific comments for tweaks to the tests, but my general note would be that the test descriptions tend to be too vague, and make it hard to verify failures.

For example, a failure of the test "should correctly determine if a card is selectable" doesn't tell me what logic is incorrect. This test specifically verifies that cards that have been already revealed cannot be selected, while unselected cards can be selected. Adding messages to each specific expectations would further help, but that might be a bit much. I'd also check with Arman and Lily to see if they agree with my suggestions, or if the specificity it isn't needd.

File: test/game_utility.test.ts (hunk 2 above)
Comment (nathanshelly) [reply test/game_utility.test.ts:39]: Will resolve in one monster commit later.

**Review by vkoves — APPROVED** [review]
Looks good to go.
